Source/DataModels/Components/ComponentAudioListener.cpp

```diff
@@ -1,60 +1,80 @@
 #include "ComponentAudioListener.h"
-#include "DataModels/Components/ComponentTransform.h"
 #include "DataModels/GameObject/GameObject.h"
+#include "DataModels/Components/ComponentTransform.h"
 
 #include "FileSystem/Json.h"
 
-ComponentAudioListener::ComponentAudioListener(const bool active, GameObject* owner) :
-	Component(ComponentType::AUDIOLISTENER, active, owner, true),
-	listenerID(owner->GetUID())
+ComponentAudioListener::ComponentAudioListener(const bool active, GameObject* owner)
+    : Component(ComponentType::AUDIOLISTENER, active, owner, true), listenerID(owner->GetUID())
 {
-	AK::SoundEngine::RegisterGameObj(listenerID, owner->GetName().c_str());
-	AK::SoundEngine::SetDefaultListeners(&listenerID, 1);
-	transform = static_cast<ComponentTransform*>(owner->GetComponent(ComponentType::TRANSFORM));
-
-	if (transform)
-	{
-		OnTransformChanged();
-	}
+    AK::SoundEngine::RegisterGameObj(listenerID, owner->GetName().c_str());
+    AK::SoundEngine::SetDefaultListeners(&listenerID, 1);
+
+    transform = static_cast<ComponentTransform*>(owner->GetComponent(ComponentType::TRANSFORM));
+
+    if (transform)
+    {
+        OnTransformChanged();
+    }
 }
 
 ComponentAudioListener::~ComponentAudioListener()
 {
-	AK::SoundEngine::UnregisterGameObj(listenerID);
+    AK::SoundEngine::UnregisterGameObj(listenerID);
 }
 
 void ComponentAudioListener::OnTransformChanged()
 {
-	const float3& pos = transform->GetGlobalPosition();
-	const float3& front = transform->GetGlobalForward();
-	const float3& correctFront = -float3(front.x, -front.y, front.z).Normalized();
-	const float3& up = transform->GetGlobalUp();
+    if (IsEnabled())
```

Comment:
Small suggestion, to avoid nesting
```suggestion
    if (!IsEnabled())
    {
        return;
    }
    // current code here
```

Reply:
If you agree with this suggestion, we could do the same in `ComponentAudioSource::OnTransformChanged` and `ComponentAudioSource::PostEvent`.